File: AgqrPlayer4Tv/AgqrPlayer4Tv/Model/Platform/ApplicationPreference.cs
using System;
using AgqrPlayer4Tv.Infrastracture;
using Android.App;
using Android.Content;
using Reactive.Bindings;
using Reactive.Bindings.Extensions;

namespace AgqrPlayer4Tv.Model.Platform
{
    public class ApplicationPreference : DisposableModelBase
    {
        public ReactiveProperty<PlayerType> PlayerType { get; } = new ReactiveProperty<PlayerType>((PlayerType)SharedPreferences.GetInt("PlayerType", (int)Platform.PlayerType.ExoPlayer));
        public ReactiveProperty<StreamingType> StreamingType { get; } = new ReactiveProperty<StreamingType>((StreamingType)SharedPreferences.GetInt("StreamingType", (int)Platform.StreamingType.Hls));
        public ReactiveProperty<bool> IsLastShutdownCorrectly { get; } = new ReactiveProperty<bool>(SharedPreferences.GetBoolean("IsLastShutdownCorrectly", true));

        private static ISharedPreferences SharedPreferences
        {
            get { return Application.Context.GetSharedPreferences("Preferences", FileCreationMode.Private); }
        }

        public ApplicationPreference()
        {
            this.PlayerType.Subscribe(x => SharedPreferences.Edit().PutInt("PlayerType", (int)x).Commit()).AddTo(this);
            this.StreamingType.Subscribe(x => SharedPreferences.Edit().PutInt("StreamingType", (int)x).Commit()).AddTo(this);
            this.IsLastShutdownCorrectly.Subscribe(x => SharedPreferences.Edit().PutBoolean("IsLastShutdownCorrectly", x).Commit()).AddTo(this);
        }
    }

    public enum PlayerType
    {
        ExoPlayer = 0,
        AndroidDefault = 1,
        WebView = 2,
    }

    public enum StreamingType
    {
        Rtmp = 0,
        Hls = 1,
    }

}

File: AgqrPlayer4Tv/app/src/main/java/org/misuzilla/agqrplayer4tv/model/preference/ApplicationPreference.kt
package org.misuzilla.agqrplayer4tv.model.preference

import android.content.Context
import android.content.SharedPreferences
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData

object ApplicationPreference {
    private val playerType = MutableLiveData(PlayerType.EXO_PLAYER)
    private val streamingType = MutableLiveData(StreamingType.HLS)
    private val isLastShutdownCorrectly = MutableLiveData(true)
    private val recommendationCurrentProgram = MutableLiveData("")

    private lateinit var context: Context
    private lateinit var sharedPreference: SharedPreferences

    fun initialize(context: Context) {
        this.context = context
        this.sharedPreference = context.getSharedPreferences("Preferences", Context.MODE_PRIVATE)

        playerType.value = PlayerType.fromInt(sharedPreference.getInt("PlayerType", PlayerType.EXO_PLAYER.value))
        streamingType.value = StreamingType.fromInt(sharedPreference.getInt("StreamingType", StreamingType.HLS.value))
        isLastShutdownCorrectly.value = sharedPreference.getBoolean("IsLastShutdownCorrectly", true)
        recommendationCurrentProgram.value = sharedPreference.getString("RecommendationCurrentProgram", "")
    }

    fun setPlayerType(value: PlayerType) {
        playerType.value = value
        sharedPreference.edit().putInt("PlayerType", value.value).commit()
    }
    fun getPlayerType(): LiveData<PlayerType> { return playerType }

    fun setStreamingType(value: StreamingType) {
        streamingType.value = value
        sharedPreference.edit().putInt("StreamingType", value.value).commit()
    }
    fun getStreamingType(): LiveData<StreamingType> { return streamingType }

    fun setIsLastShutdownCorrectly(value: Boolean) {
        isLastShutdownCorrectly.value = value
        sharedPreference.edit().putBoolean("IsLastShutdownCorrectly", value).commit()
    }
    fun getIsLastShutdownCorrectly(): LiveData<Boolean> { return isLastShutdownCorrectly }

    fun setRecommendationCurrentProgram(value: String) {
        recommendationCurrentProgram.value = value
        sharedPreference.edit().putString("RecommendationCurrentProgram", value).commit()
    }
    fun getRecommendationCurrentProgram(): LiveData<String> { return recommendationCurrentProgram }
}
